groups/bsl/bslstl/bslstl_monostate.t.cpp

```diff
@@ -0,0 +1,309 @@
+    bool verbose = argc > 2;
+    //  bool veryVerbose         = argc > 3;
+    //  bool veryVeryVerbose     = argc > 4;
+    //  bool veryVeryVeryVerbose = argc > 5;
+
+    printf("TEST " __FILE__ " CASE %d\n", test);
+
+    switch (test) {
+      case 0:
+
+      case 5: {
+        // --------------------------------------------------------------------
+        // TESTING 'hashAppend'
+        //
+        // Concerns:
+        //: 1 Hashing a monostate object is equivalent to appending 'false'
+        //:   to the hash.
+        //
+        // Plan:
+        //: 1 Create a monostate object and verify that hashing it yields the
+        //:   same value as hashing 'true'. [C-1]
+        //
+        // Testing:
+        //   void hashAppend(HASHALG& hashAlg, const monostate &);
+        // --------------------------------------------------------------------
+
+        if (verbose)
+            printf("\nTESTING 'hashAppend'"
+                   "\n====================\n");
+
+        {
+            bsl::monostate obj;
+
+            const size_t hashValue1 = bslh::Hash<>()(obj);
+            const size_t hashValue2 = bslh::Hash<>()(true);
+            ASSERTV(hashValue1, hashValue2, hashValue1 == hashValue2);
+        }
+      } break;
+
+      case 4: {
+        // --------------------------------------------------------------------
+        // 'noexcept' SPECIFICATION
+        //
+        // Concerns:
+        //: 1 The 'noexcept' specification has been applied to
+        //:   all relational operators as required by the standard.
+        //
+        // Plan:
+        //: 1 Apply each unary 'noexcept' operator to each relational
+        //:   operator, confirm that calculated boolean value matches the
+        //:   expected value.
+        //
+        // Testing:
+        //   CONCERN: Methods qualified 'noexcept' in standard are so.
+        // --------------------------------------------------------------------
+
+        if (verbose)
+            printf("\n'noexcept' SPECIFICATION"
+                   "\n========================\n");
+
+        bsl::monostate lhs;
+        bsl::monostate rhs;
+
+        ASSERT(BSLS_KEYWORD_NOEXCEPT_AVAILABLE ==
+               BSLS_KEYWORD_NOEXCEPT_OPERATOR(lhs == rhs));
+        ASSERT(BSLS_KEYWORD_NOEXCEPT_AVAILABLE ==
+               BSLS_KEYWORD_NOEXCEPT_OPERATOR(lhs != rhs));
+        ASSERT(BSLS_KEYWORD_NOEXCEPT_AVAILABLE ==
+               BSLS_KEYWORD_NOEXCEPT_OPERATOR(lhs < rhs));
+        ASSERT(BSLS_KEYWORD_NOEXCEPT_AVAILABLE ==
+               BSLS_KEYWORD_NOEXCEPT_OPERATOR(lhs > rhs));
+        ASSERT(BSLS_KEYWORD_NOEXCEPT_AVAILABLE ==
+               BSLS_KEYWORD_NOEXCEPT_OPERATOR(lhs <= rhs));
+        ASSERT(BSLS_KEYWORD_NOEXCEPT_AVAILABLE ==
+               BSLS_KEYWORD_NOEXCEPT_OPERATOR(lhs >= rhs));
+
+      } break;
+
+      case 3: {
+        // --------------------------------------------------------------------
+        // Relational operators
```

Comment:
The style guide doesn't explicitly mention this, but bde_verify needs the banners to be in ALL CAPS in order to be happy (with the exception of anything that's quoted, like `'bsl::monostate'`).
```suggestion
        // RELATIONAL OPERATORS
```